lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
> we can not simply drop the old approach as this would break existing applications and not follow semver.

I agree with that. What about a flag to remove the old behavior? I think that if we lazy-load we need to drop the old behavior to avoid the inconsistent behavior. Also, serialize is so rarely used that even for large apps we should be fine moving all at once.